src/NovelRT/Graphics/RenderObject.cpp

```diff
@@ -0,0 +1,89 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root for more information.
+
+#include <NovelRT.h>
+
+namespace NovelRT::Graphics {
+
+  void RenderObject::OnCameraViewChanged(CameraViewChangedEventArgs args) {
+    _uboCameraData = args.cameraMatrix;
+  }
+
+  RenderObject::RenderObject(const Transform& transform, int layer, ShaderProgram shaderProgram, Camera* camera) :
+    WorldObject(transform, layer),
+    _finalViewMatrixData(Utilities::Lazy<Maths::GeoMatrix4<float>>(std::function<Maths::GeoMatrix4<float>()>(std::bind(&RenderObject::generateViewData, this)))),
+    _vertexBuffer(Utilities::Lazy<GLuint>(std::function<GLuint()>(generateStandardBuffer))),
+    _vertexArrayObject(Utilities::Lazy<GLuint>(std::function<GLuint()>([] {
+    GLuint tempVao;
+    glGenVertexArrays(1, &tempVao);
+    return tempVao;
+      }))),
+    _shaderProgram(shaderProgram),
+    _camera(camera),
+    _uboCameraData(_camera->getCameraUboMatrix()) {
+    _camera->subscribeToCameraViewChanged([this](auto x) { OnCameraViewChanged(x); }); //TODO: Should we refactor the method into the lambda?
+  }
+
+  void RenderObject::executeObjectBehaviour() {
+    if (_isDirty) {
+      _finalViewMatrixData.reset();
+      _bufferInitialised = false;
+      _isDirty = false;
+    }
+
+    if (!_bufferInitialised) {
+      configureObjectBuffers();
+      _bufferInitialised = true;
+    }
+    drawObject();
+  }
+
+  void RenderObject::configureObjectBuffers() {
+    auto topLeft = Maths::GeoVector<GLfloat>(-0.5f, 0.5f);
+    auto bottomRight = Maths::GeoVector<GLfloat>(0.5f, -0.5f);
+    auto topRight = Maths::GeoVector<GLfloat>(0.5f, 0.5f);
+    auto bottomLeft = Maths::GeoVector<GLfloat>(-0.5f, -0.5f);
+    _vertexBufferData = {
```

Comment:
same thing, this could be shared across multiple instances.

Reply:
Same as before, I think we should look to do these sorts of changes alongside the buffer overhaul.